diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index f9e4c6c7..f159e2e3 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -9,6 +9,8 @@ file( COPY ncepbufr DESTINATION . )
 add_custom_target(python_mod ALL)
 add_dependencies(python_mod bufr_4)
 
+separate_arguments(MESON_ARGS NATIVE_COMMAND $ENV{MESON_ARGS})
+
 add_custom_command(
   TARGET python_mod
   POST_BUILD
@@ -17,6 +19,10 @@ add_custom_command(
                    -m _bufrlib
                    -L${CMAKE_BINARY_DIR}/src
                    -lbufr_4
+                   --backend meson
+                   --build-dir "${CMAKE_BINARY_DIR}/f2py"
+                   COMMAND meson setup ${MESON_ARGS} "${CMAKE_BINARY_DIR}/f2py"
+                   COMMAND meson build "${CMAKE_BINARY_DIR}/f2py"
   DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/_bufrlib.pyf"
 )
 
