diff --git a/python/test/test_checkpoint.py b/python/test/test_checkpoint.py
index 6c165ed2..29e22eba 100644
--- a/python/test/test_checkpoint.py
+++ b/python/test/test_checkpoint.py
@@ -15,7 +15,7 @@ while bufr.advance() == 0:
     #print(nmsg, bufr.msg_counter, bufr.msg_type, bufr.msg_date)
     while bufr.load_subset() == 0:
         hdr = bufr.read_subset(hdstr).squeeze()
-        station_id = hdr[0].tostring()
+        station_id = hdr[0].tobytes()
         obs = bufr.read_subset(obstr)
         nlevs = obs.shape[-1]
         oer = bufr.read_subset(oestr)
@@ -31,7 +31,7 @@ bufr.restore()
 # statement is no longer needed.
 #bufr.load_subset()
 hdr = bufr.read_subset(hdstr).squeeze()
-station_id = hdr[0].tostring()
+station_id = hdr[0].tobytes()
 obs = bufr.read_subset(obstr)
 nlevs = obs.shape[-1]
 oer = bufr.read_subset(oestr)
diff --git a/python/test/test_prepbufr.py b/python/test/test_prepbufr.py
index f6b4b446..92288c58 100644
--- a/python/test/test_prepbufr.py
+++ b/python/test/test_prepbufr.py
@@ -16,7 +16,7 @@ while bufr.advance() == 0: # loop over messages.
     #bufr.read_subset(obstr) # should raise subset not loaded error
     while bufr.load_subset() == 0: # loop over subsets in message.
         hdr = bufr.read_subset(hdstr).squeeze()
-        station_id = hdr[0].tostring()
+        station_id = hdr[0].tobytes()
         obs = bufr.read_subset(obstr)
         nlevs = obs.shape[-1]
         oer = bufr.read_subset(oestr)
diff --git a/python/test/test_write.py b/python/test/test_write.py
index 69140cf3..66be32a6 100644
--- a/python/test/test_write.py
+++ b/python/test/test_write.py
@@ -17,7 +17,7 @@ subset='ADPSFC'  # surface land (SYNOPTIC, METAR) reports
 bufr.open_message(subset, idate)
 
 hdr = bufr.missing_value*np.ones(len(hdstr.split()),np.float64)
-hdr[0] = np.fromstring('KTKI    ',dtype=np.float64)[0]
+hdr[0] = np.frombuffer(b'KTKI    ',dtype=np.float64)[0]
 hdr[1]=263.4; hdr[2]=33.2; hdr[3] = -0.1; hdr[4]=287; hdr[5]=179
 # encode header for wind obs
 bufr.write_subset(hdr,hdstr)
@@ -53,7 +53,7 @@ bufr.open_message(subset, idate)
 
 # set header
 hdr[:]=bufr.missing_value
-hdr[0] = np.fromstring('72293   ',dtype=np.float64)[0]
+hdr[0] = np.frombuffer(b'72293   ',dtype=np.float64)[0]
 hdr[1]=242.9; hdr[2]=32.9; hdr[3]=0.0; hdr[5]=134.0
 
 # set obs, qcf, oer for  wind
@@ -111,7 +111,7 @@ hdr = bufr.missing_value*np.ones(len(hdstr.split()),np.float64)
 obs = bufr.missing_value*np.ones(len(obstr.split()),np.float64)
 oer = bufr.missing_value*np.ones(len(oestr.split()),np.float64)
 qcf = bufr.missing_value*np.ones(len(qcstr.split()),np.float64)
-hdr[0] = np.fromstring('KBOU    ',dtype=np.float64)[0]
+hdr[0] = np.frombuffer(b'KBOU    ',dtype=np.float64)[0]
 hdr[1]=-105.0;hdr[2]=40.0;hdr[3]=-1.0;hdr[4]=181
 obs[0]=300.0
 idate=2008120101  # YYYYMMDDHH
@@ -132,7 +132,7 @@ while bufr.advance() == 0: # loop over messages.
     print(bufr.msg_counter, bufr.msg_type, bufr.msg_date)
     while bufr.load_subset() == 0: # loop over subsets in message.
         hdr = bufr.read_subset(hdstr).squeeze()
-        station_id = hdr[0].tostring()
+        station_id = hdr[0].tobytes()
         obs = bufr.read_subset(obstr)
         nlevs = obs.shape[-1]
         oer = bufr.read_subset(oestr)
diff --git a/python/utils/bufrdif.py b/python/utils/bufrdif.py
index 8fe3b9fd..e5404351 100644
--- a/python/utils/bufrdif.py
+++ b/python/utils/bufrdif.py
@@ -81,16 +81,16 @@ def get_bufr_dict(bufr,verbose=False,bufr_type='prep'):
                 secs = int(hdr[3]*3600.)
                 obdate = refdate + secs*delta
                 hdr[3]=float(obdate.strftime('%Y%m%d%H%M%S')) # YYYYMMDDHHMMSS
-            hdrhash = hash(hdr.tostring())
-            obshash = hash(bufr.read_subset(obstr).tostring())
+            hdrhash = hash(hdr.tobytes())
+            obshash = hash(bufr.read_subset(obstr).tobytes())
             if bufr_type == 'goesfv':
                 key = '%s %s %s' % (bufr.msg_type,hdrhash,obshash)
             else:
-                qchash = hash(bufr.read_subset(qcstr).tostring())
+                qchash = hash(bufr.read_subset(qcstr).tobytes())
                 key = '%s %s %s %s' % (bufr.msg_type,hdrhash,obshash,qchash)
             # human readable header string
             #if bufr_type == 'prep':
-            #    hdrstr = '%8s %8.3f %7.3f %14s %3i %5i %3i' % (hdr[0].tostring(),hdr[1],hdr[2],int(hdr[3]),int(hdr[4]),int(hdr[5]),int(hdr[6])))
+            #    hdrstr = '%8s %8.3f %7.3f %14s %3i %5i %3i' % (hdr[0].tobytes(),hdr[1],hdr[2],int(hdr[3]),int(hdr[4]),int(hdr[5]),int(hdr[6])))
             nsubset += 1
             if key in bufr_dict:
                 ndup += 1
diff --git a/python/utils/prepbufr2nc b/python/utils/prepbufr2nc
index 764cea1f..296caf17 100755
--- a/python/utils/prepbufr2nc
+++ b/python/utils/prepbufr2nc
@@ -137,7 +137,7 @@ while bufr.advance() == 0: # loop over messages.
             elif hdr[4] >= 223 and hdr[4] <= 228:
                 z = obs[3,nlev] # use zob, not station elev
             obidstr = "%s %3i %6.2f %6.2f %9.5f %5i %6.1f" % \
-            (hdr[0].tostring(), hdr[4], lon, lat, time, z, press)
+            (hdr[0].tobytes(), hdr[4], lon, lat, time, z, press)
             # skip ob if there is already a matching obid string
             # (do we need more checking here to make sure obs are
             # really duplicates?)
diff --git a/python/utils/prepbufr_rewrite b/python/utils/prepbufr_rewrite
index 07c21ca5..ab570fbb 100755
--- a/python/utils/prepbufr_rewrite
+++ b/python/utils/prepbufr_rewrite
@@ -43,7 +43,7 @@ while bufrin.advance() == 0: # loop over messages.
     while bufrin.load_subset() == 0: # loop over subsets in message.
         # read subset header from bufrin
         hdr = bufrin.read_subset(hdstr).squeeze()
-        stid = hdr[0].tostring()
+        stid = hdr[0].tobytes()
         lon = hdr[1]; lat = hdr[2]; obcode = int(hdr[3])
         # skip ENRR obs (drops from GIV,Global Hawk and C130s) and sondes (from
         # Christmas Island and Ron Brown))
